File: go/0167.two-sum-ii-input-array-is-sorted/solution.go
// Created by Olgierd Palasz at 2026/05/06 21:34
// leetgo: dev
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

package main

import (
	"bufio"
	"fmt"
	"os"

	. "github.com/j178/leetgo/testutils/go"
)

// @lc code=begin

func twoSum(numbers []int, target int) []int {
	i := 0
	j := len(numbers) - 1

	for numbers[i]+numbers[j] != target {
		if numbers[i]+numbers[j] > target {
			j -= 1
		} else {
			i += 1
		}
	}
	return []int{i + 1, j + 1}
}

// @lc code=end

func main() {
	stdin := bufio.NewReader(os.Stdin)
	numbers := Deserialize[[]int](ReadLine(stdin))
	target := Deserialize[int](ReadLine(stdin))
	ans := twoSum(numbers, target)

	fmt.Println("\noutput:", Serialize(ans))
}

File: rust/src/0167.two-sum-ii-input-array-is-sorted/solution.rs
// Created by Olgierd Palasz at 2026/07/21 23:04
// leetgo: dev
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

use anyhow::Result;
use leetgo_rs::*;

struct Solution;

// @lc code=begin
use std::ops::Add;
impl Solution {
    pub fn two_sum(numbers: Vec<i32>, target: i32) -> Vec<i32> {
        let mut left = 0;
        let mut right = numbers.len() - 1;
        while left < right {
            if numbers[left] + numbers[right] < target {
                left += 1;
            } else if numbers[left] + numbers[right] > target {
                right -= 1;
            } else {
                return vec![left.add(1_usize) as i32, right.add(1_usize) as i32];
            }
        }

        vec![]
    }
}

// @lc code=end

fn main() -> Result<()> {
    let numbers: Vec<i32> = deserialize(&read_line()?)?;
    let target: i32 = deserialize(&read_line()?)?;
    let ans: Vec<i32> = Solution::two_sum(numbers, target).into();

    println!("\noutput: {}", serialize(ans)?);
    Ok(())
}
